### PR TITLE
Avoid loading global boto config

### LINKED ISSUE #3766: Errors during Open edX migration task: Could not load Boto's S3 bindings
Hi Guys,

I set up an Open edX Ficus platform on my local physical machine and it works normally. I am trying to set up the same platform on a Google Cloud Platform (gcp) virtual machine but it fails with the following migration error:

```
TASK [edxapp : migrate] ********************************************************
failed: [localhost] (item=lms) => {"changed": true, "cmd": ["/edx/bin/edxapp-migrate-lms"], "delta": "0:00:02.032571", "end": "2017-03-21 07:52:27.582931", "failed": true, "item": "lms", "rc": 1, "start": "2017-03-21 07:52:25.550360", "stderr": "Traceback (most recent call last):\n  File \"manage.py\", line 111, in <module>\n    startup = importlib.import_module(edx_args.startup)\n  File \"/usr/lib/python2.7/importlib/__init__.py\", line 37, in import_module\n    __import__(name)\n  File \"/edx/app/edxapp/edx-platform/lms/startup.py\", line 21, in <module>\n    import lms_xblock.runtime\n  File \"/edx/app/edxapp/edx-platform/lms/djangoapps/lms_xblock/runtime.py\", line 9, in <module>\n    from openedx.core.djangoapps.user_api.course_tag import api as user_course_tag_api\n  File \"/edx/app/edxapp/edx-platform/openedx/core/djangoapps/user_api/course_tag/api.py\", line 10, in <module>\n    from ..models import UserCourseTag\n  File \"/edx/app/edxapp/edx-platform/openedx/core/djangoapps/user_api/models.py\", line 20, in <module>\n    from student.models import UserProfile, Registration, PendingEmailChange  # pylint: disable=unused-import\n  File \"/edx/app/edxapp/edx-platform/common/djangoapps/student/models.py\", line 49, in <module>\n    from certificates.models import GeneratedCertificate\n  File \"/edx/app/edxapp/edx-platform/lms/djangoapps/certificates/__init__.py\", line 3, in <module>\n    from . import signals\n  File \"/edx/app/edxapp/edx-platform/lms/djangoapps/certificates/signals.py\", line 7, in <module>\n    from certificates.models import CertificateGenerationCourseSetting\n  File \"/edx/app/edxapp/edx-platform/lms/djangoapps/certificates/models.py\", line 69, in <module>\n    from lms.djangoapps.instructor_task.models import InstructorTask\n  File \"/edx/app/edxapp/edx-platform/lms/djangoapps/instructor_task/models.py\", line 26, in <module>\n    from openedx.core.storage import get_storage\n  File \"/edx/app/edxapp/edx-platform/openedx/core/storage.py\", line 16, in <module>\n    from storages.backends.s3boto import S3BotoStorage\n  File \"/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/storages/backends/s3boto.py\", line 19, in <module>\n    raise ImproperlyConfigured(\"Could not load Boto's S3 bindings.\\n\"\ndjango.core.exceptions.ImproperlyConfigured: Could not load Boto's S3 bindings.\nSee https://github.com/boto/boto", "stdout": "", "stdout_lines": [], "warnings": []}
failed: [localhost] (item=cms) => {"changed": true, "cmd": ["/edx/bin/edxapp-migrate-cms"], "delta": "0:00:02.579207", "end": "2017-03-21 07:52:30.266621", "failed": true, "item": "cms", "rc": 1, "start": "2017-03-21 07:52:27.687414", "stderr": "Traceback (most recent call last):\n  File \"manage.py\", line 112, in <module>\n    startup.run()\n  File \"/edx/app/edxapp/edx-platform/cms/startup.py\", line 39, in run\n    django.setup()\n  File \"/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/__init__.py\", line 18, in setup\n    apps.populate(settings.INSTALLED_APPS)\n  File \"/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/apps/registry.py\", line 85, in populate\n    app_config = AppConfig.create(entry)\n  File \"/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/apps/config.py\", line 86, in create\n    module = import_module(entry)\n  File \"/usr/lib/python2.7/importlib/__init__.py\", line 37, in import_module\n    __import__(name)\n  File \"/edx/app/edxapp/venvs/edxapp/src/edx-sga/edx_sga/__init__.py\", line 1, in <module>\n    from .sga import StaffGradedAssignmentXBlock\n  File \"/edx/app/edxapp/venvs/edxapp/src/edx-sga/edx_sga/sga.py\", line 23, in <module>\n    from student.models import user_by_anonymous_id\n  File \"/edx/app/edxapp/edx-platform/common/djangoapps/student/models.py\", line 49, in <module>\n    from certificates.models import GeneratedCertificate\n  File \"/edx/app/edxapp/edx-platform/lms/djangoapps/certificates/__init__.py\", line 3, in <module>\n    from . import signals\n  File \"/edx/app/edxapp/edx-platform/lms/djangoapps/certificates/signals.py\", line 7, in <module>\n    from certificates.models import CertificateGenerationCourseSetting\n  File \"/edx/app/edxapp/edx-platform/lms/djangoapps/certificates/models.py\", line 69, in <module>\n    from lms.djangoapps.instructor_task.models import InstructorTask\n  File \"/edx/app/edxapp/edx-platform/lms/djangoapps/instructor_task/models.py\", line 26, in <module>\n    from openedx.core.storage import get_storage\n  File \"/edx/app/edxapp/edx-platform/openedx/core/storage.py\", line 16, in <module>\n    from storages.backends.s3boto import S3BotoStorage\n  File \"/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/storages/backends/s3boto.py\", line 19, in <module>\n    raise ImproperlyConfigured(\"Could not load Boto's S3 bindings.\\n\"\ndjango.core.exceptions.ImproperlyConfigured: Could not load Boto's S3 bindings.\nSee https://github.com/boto/boto", "stdout": "", "stdout_lines": [], "warnings": []}

RUNNING HANDLER [mongo : restart mongo] ****************************************
	to retry, use: --limit @/var/tmp/configuration/playbooks/edx_sandbox.retry

PLAY RECAP *********************************************************************
localhost                  : ok=196  changed=130  unreachable=0    failed=1   

INFO:/var/tmp/configuration/playbooks/callback_plugins/task_timing: install python requirements --------------------------------------------------- 613.02s
INFO:/var/tmp/configuration/playbooks/callback_plugins/task_timing: code sandbox | Install base sandbox requirements and create sandbox virtualenv  226.86s
INFO:/var/tmp/configuration/playbooks/callback_plugins/task_timing: code sandbox | Install sandbox requirements into sandbox venv ------------------ 57.36s
INFO:/var/tmp/configuration/playbooks/callback_plugins/task_timing: checkout edx-platform repo into /edx/app/edxapp/edx-platform ------------------- 51.81s
INFO:/var/tmp/configuration/playbooks/callback_plugins/task_timing: Install python-software-properties if debian ----------------------------------- 43.73s
INFO:/var/tmp/configuration/playbooks/callback_plugins/task_timing: Install mysql-5.6 and dependencies --------------------------------------------- 21.89s
INFO:/var/tmp/configuration/playbooks/callback_plugins/task_timing: Install mongo server and recommends -------------------------------------------- 14.63s
INFO:/var/tmp/configuration/playbooks/callback_plugins/task_timing: Install packages needed for single server -------------------------------------- 11.70s
INFO:/var/tmp/configuration/playbooks/callback_plugins/task_timing: Install rabbit package using gdebi --------------------------------------------- 10.81s
INFO:/var/tmp/configuration/playbooks/callback_plugins/task_timing: install node dependencies ------------------------------------------------------ 10.61s
INFO:/var/tmp/configuration/playbooks/callback_plugins/task_timing:
Playbook Configure instance(s) finished: 2017-03-21 07:52:30.282508, 343 total tasks.  0:20:13.652374 elapsed. 
```
The  stack trace above shows the problem `Could not load Boto's S3 bindings.`:

```
LMS:
Traceback (most recent call last):\n  
File \"manage.py\", line 111, in <module>\n    startup = importlib.import_module(edx_args.startup)\n  File \"/usr/lib/python2.7/importlib/__init__.py\", line 37, in import_module\n    __import__(name)\n  
File \"/edx/app/edxapp/edx-platform/lms/startup.py\", line 21, in <module>\n    import lms_xblock.runtime\n  
File \"/edx/app/edxapp/edx-platform/lms/djangoapps/lms_xblock/runtime.py\", line 9, in <module>\n    from openedx.core.djangoapps.user_api.course_tag import api as user_course_tag_api\n  
File \"/edx/app/edxapp/edx-platform/openedx/core/djangoapps/user_api/course_tag/api.py\", line 10, in <module>\n    from ..models import UserCourseTag\n  
File \"/edx/app/edxapp/edx-platform/openedx/core/djangoapps/user_api/models.py\", line 20, in <module>\n    from student.models import UserProfile, Registration, PendingEmailChange  # pylint: disable=unused-import\n  
File \"/edx/app/edxapp/edx-platform/common/djangoapps/student/models.py\", line 49, in <module>\n    from certificates.models import GeneratedCertificate\n  
File \"/edx/app/edxapp/edx-platform/lms/djangoapps/certificates/__init__.py\", line 3, in <module>\n    from . import signals\n  
File \"/edx/app/edxapp/edx-platform/lms/djangoapps/certificates/signals.py\", line 7, in <module>\n    from certificates.models import CertificateGenerationCourseSetting\n  
File \"/edx/app/edxapp/edx-platform/lms/djangoapps/certificates/models.py\", line 69, in <module>\n    from lms.djangoapps.instructor_task.models import InstructorTask\n  
File \"/edx/app/edxapp/edx-platform/lms/djangoapps/instructor_task/models.py\", line 26, in <module>\n    from openedx.core.storage import get_storage\n  
File \"/edx/app/edxapp/edx-platform/openedx/core/storage.py\", line 16, in <module>\n    from storages.backends.s3boto import S3BotoStorage\n  
File \"/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/storages/backends/s3boto.py\", line 19, in <module>\n    
raise ImproperlyConfigured(\"Could not load Boto's S3 bindings.\\n\"\ndjango.core.exceptions.ImproperlyConfigured: Could not load Boto's S3 bindings.\nSee https://github.com/boto/boto"

CMS:
Traceback (most recent call last):\n  
File \"manage.py\", line 112, in <module>\n    startup.run()\n 
File \"/edx/app/edxapp/edx-platform/cms/startup.py\", line 39, in run\n    django.setup()\n 
File \"/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/__init__.py\", line 18, in setup\n    apps.populate(settings.INSTALLED_APPS)\n  
File \"/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/apps/registry.py\", line 85, in populate\n    app_config = AppConfig.create(entry)\n  
File \"/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/apps/config.py\", line 86, in create\n    module = import_module(entry)\n  
File \"/usr/lib/python2.7/importlib/__init__.py\", line 37, in import_module\n    __import__(name)\n 
File \"/edx/app/edxapp/venvs/edxapp/src/edx-sga/edx_sga/__init__.py\", line 1, in <module>\n    from .sga import StaffGradedAssignmentXBlock\n 
File \"/edx/app/edxapp/venvs/edxapp/src/edx-sga/edx_sga/sga.py\", line 23, in <module>\n    from student.models import user_by_anonymous_id\n  
File \"/edx/app/edxapp/edx-platform/common/djangoapps/student/models.py\", line 49, in <module>\n    from certificates.models import GeneratedCertificate\n  
File \"/edx/app/edxapp/edx-platform/lms/djangoapps/certificates/__init__.py\", line 3, in <module>\n    from . import signals\n  
File \"/edx/app/edxapp/edx-platform/lms/djangoapps/certificates/signals.py\", line 7, in <module>\n    from certificates.models import CertificateGenerationCourseSetting\n  
File \"/edx/app/edxapp/edx-platform/lms/djangoapps/certificates/models.py\", line 69, in <module>\n    from lms.djangoapps.instructor_task.models import InstructorTask\n  
File \"/edx/app/edxapp/edx-platform/lms/djangoapps/instructor_task/models.py\", line 26, in <module>\n    from openedx.core.storage import get_storage\n  
File \"/edx/app/edxapp/edx-platform/openedx/core/storage.py\", line 16, in <module>\n    from storages.backends.s3boto import S3BotoStorage\n  
File \"/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/storages/backends/s3boto.py\", line 19, in <module>\n    
raise ImproperlyConfigured(\"Could not load Boto's S3 bindings.\\n\"\ndjango.core.exceptions.ImproperlyConfigured: Could not load Boto's S3 bindings.\nSee https://github.com/boto/boto"
```

The installation steps I took for the vm are:
```
$ sudo apt-get update -y
$ sudo apt-get upgrade -y
$ sudo reboot

$ export OPENEDX_RELEASE=open-release/ficus.1
$ wget https://raw.githubusercontent.com/edx/configuration/$OPENEDX_RELEASE/util/install/ansible-bootstrap.sh -O - | sudo bash
$ source /edx/app/edx_ansible/venvs/edx_ansible/bin/activate
$ wget https://raw.githubusercontent.com/edx/configuration/$OPENEDX_RELEASE/util/install/sandbox.sh -O - | bash
```
The gcp vm is an Ubuntu 16.04 LTS Xenial server machine with the following parameters: 2 vCPUs, 8 GB memory, 100 GB storage.

Any suggestion how to solve this ?

### PR DESCRIPTION
**Pull request by mrgnr**
By default, boto loads a global configuration file from `/etc/boto.cfg`. On GCP, this file defines options we don't want to load, like `plugin_directory`. We can avoid loading the global config by creating a local (empty) [`.boto`](http://boto.cloudhackers.com/en/latest/boto_config_tut.html) file and load it by setting the `BOTO_CONFIG` environment variable.

Fixes #3766

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @mrgnr! I've created [OSPR-1729](https://openedx.atlassian.net/browse/OSPR-1729) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



If you like, you can add yourself to the [AUTHORS](https://github.com/edx/configuration/blob/master/AUTHORS)      file for this repo, though that isn't required.     Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

**Comment by mrgnr** [comment]
@gsong The issue fixed by this PR will affect anyone deploying on GCP. Could we have someone review these changes please?

**Comment by mrgnr** [comment]
Ping @gsong

**Comment by gsong** [comment]
@edx/devops Ping.

File: playbooks/roles/edxapp/templates/cms.conf.j2 [hunk 1]
```diff
@@ -10,7 +10,7 @@ command={{ executable }} -c {{ edxapp_app_dir }}/cms_gunicorn.py {{ EDXAPP_CMS_G
 
 user={{ common_web_user }}
 directory={{ edxapp_code_dir }}
-environment={% if COMMON_ENABLE_NEWRELIC_APP %}NEW_RELIC_APP_NAME={{ EDXAPP_NEWRELIC_CMS_APPNAME }},NEW_RELIC_LICENSE_KEY={{ NEWRELIC_LICENSE_KEY }},{% endif -%}PORT={{ edxapp_cms_gunicorn_port }},ADDRESS={{ edxapp_cms_gunicorn_host }},LANG={{ EDXAPP_LANG }},DJANGO_SETTINGS_MODULE={{ EDXAPP_CMS_ENV }},SERVICE_VARIANT="cms"
+environment={% if COMMON_ENABLE_NEWRELIC_APP %}NEW_RELIC_APP_NAME={{ EDXAPP_NEWRELIC_CMS_APPNAME }},NEW_RELIC_LICENSE_KEY={{ NEWRELIC_LICENSE_KEY }},{% endif -%}PORT={{ edxapp_cms_gunicorn_port }},ADDRESS={{ edxapp_cms_gunicorn_host }},LANG={{ EDXAPP_LANG }},DJANGO_SETTINGS_MODULE={{ EDXAPP_CMS_ENV }},SERVICE_VARIANT="cms",BOTO_CONFIG="{{ edxapp_app_dir }}/.boto"
```
Comment (nadeemshahzad) [inline playbooks/roles/edxapp/templates/cms.conf.j2]: @mrgnr I think BOTO_CONFIG variable needs to be defined in  file "/edx/app/edxapp/edxapp_env"  as it is sourced by the migration script. Can you please make sure that the template "edxapp_env.j2" is rendered with BOTO_CONFIG variable and its corresponding value?

File: playbooks/roles/notifier/defaults/main.yml [hunk 2]
```diff
@@ -113,3 +113,4 @@ notifier_env_vars:
   FORUM_DIGEST_TASK_BATCH_SIZE: "{{ NOTIFIER_FORUM_DIGEST_TASK_BATCH_SIZE }}"
   FORUM_DIGEST_TASK_RATE_LIMIT: "{{ NOTIFIER_FORUM_DIGEST_TASK_RATE_LIMIT }}"
   DEAD_MANS_SNITCH_URL: "{{ NOTIFIER_DEAD_MANS_SNITCH_URL }}"
+  BOTO_CONFIG: "{{ notifier_app_dir }}/.boto"
```
Comment (nadeemshahzad) [inline playbooks/roles/notifier/defaults/main.yml]: @mrgnr Same as my comment on edxapp role.

**Comment by nadeemshahzad** [comment]
I have done the basic testing 
Created sandbox with changes and verified things are working as expected.
Confirmed boto config issue on google compute platform and verified that fix in this PR is working.
Changes Looks good to me.

**Comment by feanil** [comment]
@nadeemshahzad did you test that things like s3 uploads and grade reports work as expected on the sandbox?

**Comment by nadeemshahzad** [comment]
@feanil I have verified s3 operations with grade report work on sandbox and corresponding objects on s3 bucket as well. All looks good to me.

**Review by nadeemshahzad — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@mrgnr 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/configuration&entry.867055334=https://github.com/edx/configuration/pull/3792&entry.1484655318=https://github.com/mrgnr&entry.752974735=2017-04-04+13:09&entry.1917517419=2017-11-06+13:17&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by devalih** [comment]
👍

**Pull request merged**